### PR TITLE
fix: 0 value handling for fraction and percentage

### LINKED ISSUE #498: gridTemplateRows 0fr is being compiled to 0px, which breaks css transition
### Describe the issue

Hey 👋

If I use gridTemplateRows with the value of `0fr`, this value gets replaced with `0px`. 

There's a [popular css trick](https://www.stefanjudis.com/snippets/how-to-animate-height-with-css-grid/), in order to smoothly open an accordeon, you may use a css transition from `gridTemplateRows: 0fr` to `1fr`. This css transition breaks if 0fr is being compiled to 0px, as px can not be transitioned to fr (i think?).

### Expected behavior

0fr shouldn't get compiled to 0px, and should be preserved. Basically, every transitionable property should be opted out from this optimization.

### Steps to reproduce

1. Create an example that uses the `gridTemplateRows` css transition trick from `0fr` to `1fr`, you may use the [mentioned article](https://www.stefanjudis.com/snippets/how-to-animate-height-with-css-grid/) as an example
2. Make sure if it works with transitioning from `repeat(1, 0fr)` to `repeat(1, 1fr)`
3. Replace `repeat(1, 0fr)` to `0fr` and `repeat(1, 1fr)` to `1fr`
4. Transition will break

### Test case

_No response_

### Additional comments

My current workaround is to use `repeat(1, 0fr)` instead of `0fr` and `repeat(1, 1fr)` insead of `1fr`

### PR DESCRIPTION
**Pull request by tounsoo**
## What changed / motivation ?

Bug fix  🐛 🔧 

## Linked PR/Issues

Fixes #498 #478 

## Additional Context

Added a test in `convert-to-className-test.js`

## Pre-flight checklist

- [x] I have read the contributing guidelines
      [Contribution Guidelines](https://github.com/facebook/stylex/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

File: packages/shared/src/utils/normalizers/zero-dimensions.js [hunk 1]
```diff
@@ -43,6 +44,8 @@ export default function normalizeZeroDimensions(
       node.value = '0deg';
     } else if (timings.indexOf(dimension.unit) !== -1) {
       node.value = '0s';
+    } else if (dimension.unit === fraction) {
+        node.value = '0fr';
```
Comment (nmn) [inline packages/shared/src/utils/normalizers/zero-dimensions.js:48]: Would you mind updating this to account for `0%` as well?

**Review by nmn — APPROVED** [review]
LGTM. Will wait a day if you're willing to fix a bit more!

File: packages/shared/src/utils/normalizers/zero-dimensions.js (hunk 1 above)
Comment (tounsoo) [reply packages/shared/src/utils/normalizers/zero-dimensions.js:49]: Done!

**Comment by tounsoo** [comment]
@nmn I also found that the contribution guide could be improved as I wasn't running the `prettier` ahead of commit.

Potentially changing the fourth point to `4. Ensure it is ready for commit (`npm run precommit`)` could improve contributing experience.

If you agree, I can make that small change on this PR or make a new PR for it. LMK 👍

**Review by nmn — APPROVED** [review]
Awesome! Thanks.

**Pull request merged**